rust/ql/src/queries/summary/TaintReach.qll

```diff
@@ -0,0 +1,31 @@
+/**
+ * Taint reach computation. Taint reach is the proportion of all dataflow nodes that can be reached
+ * via taint flow from any active thread model source. It's usually expressed per million nodes.
+ */
+
+import rust
+private import codeql.rust.Concepts
+private import codeql.rust.dataflow.DataFlow
+private import codeql.rust.dataflow.TaintTracking
+
+/**
+ * A taint configuration for taint reach (flow to any node from any modeled source).
+ */
+private module TaintReachConfig implements DataFlow::ConfigSig {
+  predicate isSource(DataFlow::Node node) { node instanceof ActiveThreatModelSource }
+
+  predicate isSink(DataFlow::Node node) { any() }
```

Comment:
This certainly looks like something that will not perform very well...

Reply:
Yes, though:
- we don't compute a path graph for this config.
- we never had any real issues with the same code in Swift.
- even on our largest database (`windows-rs`) with a slightly warmed up cache, quick eval-ing `getTaintReach()` takes 2 seconds.  With vast amounts of fake sources etc added to increase reach beyond what I think is ever plausible, 24s.

I'm guessing execution time is roughly linear in the number of nodes reached???  I'd be interested to hear your thoughts, concerns and suggestions - even if this means changing what we measure.

Reply:
OK, let's leave it as-is for now then.